manifests/database.pp

```diff
@@ -14,7 +14,7 @@
   if $foreman::db_manage_rake {
     $seed_env = {
       'SEED_ADMIN_USER'       => $foreman::initial_admin_username,
-      'SEED_ADMIN_PASSWORD'   => $foreman::initial_admin_password,
+      'SEED_ADMIN_PASSWORD'   => if $foreman::initial_admin_password =~ Sensitive {$foreman::initial_admin_password.unwrap} else {$foreman::initial_admin_password},
```

Comment:
How about a private `foreman::unwrap_sensitive` function?
```puppet
# @api private
function foreman::unwrap_sensitive ($value) {
  if $value =~ Sensitive {
    $value.unwrap
  } else {
    $value
  }
}
```

Reply:
Not necessary anymore with [this Pullrequest](https://github.com/puppetlabs/puppet/pull/8651/files).

Reply:
I'm leaning to see when the Puppet core PR will be merged since there appears to be agreement on the general design already. Perhaps we can wait for a Puppet version with improved unwrap and set that as the minimum version? It'll keep the code cleaner.

Reply:
[First one](https://github.com/theforeman/puppet-foreman/pull/962) of the split-off and separated Pullrequests.

Reply:
> Not necessary anymore with [this Pullrequest](https://github.com/puppetlabs/puppet/pull/8651/files).

Just to close this discussion: this is available in Puppet 6.24.0 but I feel like requiring that version is a bit too early.  So let's stick with the long form for now.